src/main/java/no/nav/familie/integrasjoner/oppgave/OppgaveService.kt

```diff
@@ -234,15 +234,20 @@ class OppgaveService constructor(
         return finnMapper(finnMappeRequest).mapper
     }
 
-    fun tilordneEnhet(
+    fun tilordneEnhetOgNullstillTilordnetRessurs(
         oppgaveId: Long,
         enhet: String,
         fjernMappeFraOppgave: Boolean,
+        nullstillTilordnetRessurs: Boolean,
         versjon: Int?,
     ) {
         val oppgave = oppgaveRestClient.finnOppgaveMedId(oppgaveId)
         val mappeId = if (fjernMappeFraOppgave) null else oppgave.mappeId
         oppgaveRestClient.oppdaterEnhet(OppgaveByttEnhet(oppgaveId, enhet, versjon ?: oppgave.versjon!!, mappeId))
+
+        if (nullstillTilordnetRessurs) {
+            oppgaveRestClient.oppdaterOppgave(Oppgave(id = oppgaveId, tilordnetRessurs = null))
```

Comment:
Her har vi nok misforstått hverandre. Se på patching av mappeId. Den har en spesial DTO
```
data class OppgaveByttEnhet(
    val id: Long,
    val tildeltEnhetsnr: String,
    val versjon: Int,
    @JsonInclude(JsonInclude.Include.ALWAYS) val mappeId: Long? = null,
)
```
Den felles objectMapperen til familie sletter alle felter som er null før man gjør en POST. Man trenger derfor å spesiallage en egen tjenste for å få slettet dette feltet, eller fikset PATCH endepunktet til å ikke slette null felt. Det siste er mer skummelt, for da må man ha kontroll på alle konsumentene gjør ting riktig. Så for å få trenger man @JsonInclude(JsonInclude.Include.ALWAYS) foran tilordetRessurs.

Reply:
Og anbefaler at man tester at tjenesten gjør det man forventer fra swaggeren til familie-integrasjoner

Reply:
Ahh takk for avklaringen Stig!

Siden det å bytte enhet nå krever en nullstilling av tilordnet ressurs etter endringer som Oppgave har gjort, så er det kanskje greit å utvide dataklassen?

Noe ala
`data class OppgaveByttEnhetOgTilordnetRessurs(
    val id: Long,
    val tildeltEnhetsnr: String,
    val versjon: Int,
    @JsonInclude(JsonInclude.Include.ALWAYS) val mappeId: Long? = null,
    @JsonInclude(JsonInclude.Include.ALWAYS) val tilordnetRessurs: String? = null,
    )
`?

Reply:
Vil funke med å utvide dataklassen og hente ut eksisterende tilordnetRessurs fra oppgaven hvis man ikke skal slette den. Hvis man skal slette den så setter man null

Reply:
Ikkesant!

Det er gjort nå! 👍